controllers/clusters/on_premise.go

```diff
@@ -0,0 +1,765 @@
+package clusters
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	k8scorev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	virtcorev1 "kubevirt.io/api/core/v1"
+	cdiv1beta1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/instaclustr/operator/apis/clusters/v1beta1"
+	"github.com/instaclustr/operator/pkg/instaclustr"
+	"github.com/instaclustr/operator/pkg/models"
+)
+
+type onPremiseBootstrap struct {
+	IcAdminAPI            instaclustr.IcadminAPI
+	K8sClient             client.Client
+	K8sObject             client.Object
+	ClusterID             string
+	CdcID                 string
+	OnPremisesSpec        *v1beta1.OnPremisesSpec
+	ExposeNodePorts       []k8scorev1.ServicePort
```

Comment:
ExposeNodePorts -> ExposePorts or PortsToExpose, let's rename because the `ExposeNodePorts` name creates wrong connotations with k8s `NodePort` service

Reply:
renamed to ExposePorts